scripts/properties.sh

```diff
@@ -43,7 +43,7 @@ TERMUX_CONFIG_PREFIX_DIR_PATH="${TERMUX_ETC_PREFIX_DIR_PATH}/termux"
 TERMUX_BOOTSTRAP_CONFIG_DIR_PATH="${TERMUX_CONFIG_PREFIX_DIR_PATH}/bootstrap"
 
 # Path to CGCT tools
-export CGCT_DIR="/data/data/${TERMUX_APP_PACKAGE}/cgct"
+export CGCT_DIR="/data/data/com.termux/cgct"
```

Comment:
Why hardcode com.termux here?

Reply:
Also here.

https://github.com/termux/termux-packages/blob/8ee01c7c8fdd38f246a265500dd8b1fa5bfd515e/clean.sh#L51

https://github.com/termux/termux-packages/commit/a46e3e9357709b7b4deb1e68ab5b4c8aba0e3d08

(`properties.sh` would need to be sourced first to use a variable)

Reply:
> Why hardcode com.termux here?

Because cgct has a hard prefix `/data/data/com.termux/cgct`, which has nothing to do with the prefix for compiling packages. But the build script is configured so that when the value of the `TERMUX_APP_PACKAGE` variable changes, the path to cgct changes, which in turn causes an error.

Reply:
What do you mean it has a hard prefix? Is it impossible for forks to compile and use cgct for their app?

Reply:
The `CGCT_DIR` variable seems to be used [here](https://github.com/termux-pacman/glibc-packages/blob/9734ddea1d1234611aa84b2287745fdc826db26d/cgct/glibc-cgct/setup-cgct#L3) and the `CGCT_APP_PREFIX` variable [here](https://github.com/termux-pacman/glibc-packages/blob/9734ddea1d1234611aa84b2287745fdc826db26d/cgct/cgt/cgct-app-prefix.cc).

Reply:
> What do you mean it has a hard prefix?

I mean that cgct is compiled with its own prefix, which does not change in any way and cannot be changed in any way through the shell.

> Is it impossible for forks to compile and use cgct for their app?

If you mean compiling cgct for forks, then yes they can do it but it will be too complicated for them. It will be much easier to use ready cgct with a configured variable `CGCT_APP_PREFIX` which should store the path (prefix) of the glibc environment of the application.

> The CGCT_DIR variable seems to be used [here](https://github.com/termux-pacman/glibc-packages/blob/9734ddea1d1234611aa84b2287745fdc826db26d/cgct/glibc-cgct/setup-cgct#L3)

The `CGCT_DIR` variable of the `setup-cgct` script is not related in any way to the `CGCT_DIR` variable of the build scripts - https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_setup_toolchain.sh#L22

Reply:
> Open source projects should provide an easy way to recompile projects including a way to change default config, and/or provide instructions at least

You say the right things, but with this logic for forks rebuilding cgct with a changed application prefix does not make sense. For them it will be a useless complication, since their final cgct will not differ in any way from the ready cgct with the `CGCT_APP_PREFIX` variable in the shell.

*PS: Compiling cgct packages is quite easy because the steps to start compiling are very similar to the steps for compiling termux packages, but I haven't made instructions for compiling cgct.*

> you should not expect everyone to use your pre-built binaries

That is, you also assume that in some fork a separate android-sdk is compiled for this fork in order to compile packages for the fork's application? /j

*PS: maybe I should rebrand cgct so people stop thinking it's only for Termux.*

Reply:
> since their final cgct will not differ in any way from the ready cgct with the CGCT_APP_PREFIX variable in the shell.

It does not necessarily need to differ. Giving people the source and giving people the ability to compile the source are equally important. Not everyone may want to use your prebuilt binaries and they may want to build the binaries in their own trusted environment, like say how F-Droid builds our apps on their own servers. A case for reproducible builds can also be made.

> I haven't made instructions for compiling cgct.

Well, you can do that in future.

>PS: maybe I should rebrand cgct so people stop thinking it's only for Termux.

That's your choice. But you only get 1 rename ;)

Reply:
> It does not necessarily need to differ. Giving people the source and giving people the ability to compile the source are equally important. Not everyone may want to use your prebuilt binaries and they may want to build the binaries in their own trusted environment. A case for reproducible builds can also be made.

Well, that's another topic with other points. 

> like say how F-Droid builds our apps on their own servers

I don't plan to publish cgct on F-Droid anytime soon `:P`